src/test/java/teammates/logic/core/FeedbackResponseCommentsLogicTest.java

```diff
@@ -346,6 +374,118 @@ public void testGetFeedbackResponseCommentsForQuestionInSection_withSectionName_
         assertEquals(0, comments.size());
     }
 
+    @Test
+    public void testIsNameVisibleToUser() {
+        FeedbackResponseCommentAttributes comment = dataBundle.feedbackResponseComments.get(COMMENT1_FROM_T1C1_TO_R1Q1S1C1);
+        FeedbackResponseAttributes relatedResponse = dataBundle.feedbackResponses.get(RESPONSE1_FOR_Q1S1C1);
+        CourseRoster roster = new CourseRoster(new ArrayList<>(courseRosterDataBundle.students.values()),
+                new ArrayList<>(courseRosterDataBundle.instructors.values()));
+
+        ______TS("success: the list that comment giver's name shown to is null; always return true");
+        comment.showGiverNameTo = null;
+        assertTrue(frcLogic.isNameVisibleToUser(comment, relatedResponse, "", roster));
+
+        ______TS("success: comment's visibility follows feedback question; always return ture");
+        comment.isVisibilityFollowingFeedbackQuestion = true;
+        assertTrue(frcLogic.isNameVisibleToUser(comment, relatedResponse, "", roster));
+
+        ______TS("success: comment is always visible to its giver");
+        assertTrue(frcLogic.isNameVisibleToUser(comment, relatedResponse, INSTRUCTOR1_IN_COURSE1_EMAIL, roster));
+
+        comment = dataBundle.feedbackResponseComments.get(COMMENT1_FROM_T1C1_TO_R1Q1S2C2);
+        assertTrue(frcLogic.isNameVisibleToUser(comment, relatedResponse, INSTRUCTOR3_IN_UNKNOWN_COURSE_EMAIL, roster));
+
+        // test user's participant type is in the list that comment giver's name is shown to
+        comment = dataBundle.feedbackResponseComments.get(COMMENT1_FROM_T1C1_TO_R1Q2S1C1);
+        relatedResponse = dataBundle.feedbackResponses.get(RESPONSE1_FOR_Q2S1C1);
+        relatedResponse.giver = STUDENT1_IN_COURSE1_EMAIL;
+
+        ______TS("success: comment is only visible to instructors");
+        comment.commentGiver = STUDENT6_IN_COURSE1_EMAIL;
+        comment.commentGiverType = STUDENTS;
+        comment.showGiverNameTo = Arrays.asList(INSTRUCTORS);
+        assertTrue(frcLogic.isNameVisibleToUser(comment, relatedResponse, INSTRUCTOR2_IN_COURSE1_EMAIL, roster));
+        assertFalse(frcLogic.isNameVisibleToUser(comment, relatedResponse, STUDENT5_IN_COURSE1_EMAIL, roster));
+        assertFalse(frcLogic.isNameVisibleToUser(comment, relatedResponse, STUDENT1_IN_COURSE1_EMAIL, roster));
+        assertFalse(frcLogic.isNameVisibleToUser(comment, relatedResponse, STUDENT3_IN_COURSE1_EMAIL, roster));
+        assertFalse(frcLogic.isNameVisibleToUser(comment, relatedResponse, STUDENT1_IN_COURSE2_EMAIL, roster));
+
+        ______TS("success: comment is only visible to response recipient");
+        comment.commentGiver = STUDENT4_IN_COURSE1_EMAIL;
+        comment.commentGiverType = STUDENTS;
+        comment.showGiverNameTo = Arrays.asList(RECEIVER);
+        assertFalse(frcLogic.isNameVisibleToUser(comment, relatedResponse, INSTRUCTOR2_IN_COURSE1_EMAIL, roster));
```

Comment:
There is no reason to test so many false cases. Check if they are really testing different behaviour and if so, leave it in.

Refer to "Equivalence Partitions" sections here: https://nus-cs2103-ay1920s2.github.io/website/se-book-adapted/chapters/testCaseDesign.html

```
By dividing possible inputs into equivalence partitions we can:
- avoid testing too many inputs from one partition. Testing too many inputs from the same partition is unlikely to find new bugs. This increases the efficiency of testing by reducing redundant test cases.
- ensure all partitions are tested. Missing partitions can result in bugs going unnoticed. This increases the effectiveness of testing by increasing the chance of finding bugs.
```

Reply:
From line 403 and below, each of the test segmentation is testing for each case in `FeedbackResponseCommentsLogic::isFeedbackParticipantNameVisibleToUser`. And each of the assertion under each test segmentation is testing for one of the following groups: instructor, response receiver, response receiver's teammate, response giver, response giver's teammate and other students. I believe here each partition is referring to each case? And in each partition, testing for all the groups of people seems to be necessary and for each group of people I only use one input to avoid inefficiency.

Reply:
That is an interesting way to look at. If you notice these lines:
```
        for (FeedbackParticipantType type : showNameTo) {
            switch (type) {
            case INSTRUCTORS:
                if (roster.getInstructorForEmail(userEmail) != null) {
                    return true;
                }
                break;
            case OWN_TEAM_MEMBERS:
                if (responseGiverTeam.equals(currentUserTeam)) {
                    return true;
                }
                break;
                ....
 ```
 
They only return True. So by testing all the true branches, I would have tested these cases. I would then need exactly one test where it matches none of the branches to return False.

Reply:
Ok I see. I will make changes for that.